File: 1-iniciante/1040/1040.lua
n1, n2, n3, n4 = io.read("*n","*n","*n","*n")

med = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / 10
print ("Media: " .. string.format("%.1f", med))

if med >= 7 then print("Aluno aprovado.")
elseif med < 5 then print("Aluno reprovado.")
elseif 5 <= med and med < 7 then
	print("Aluno em exame.")

	n5 = io.read("*n")
	print("Nota do exame: " .. string.format("%.1f", n5))

	med = (med + n5) / 2

	if med >= 5 then print("Aluno aprovado.")
	elseif med < 5 then print("Aluno reprovado.") end
	print ("Media final: " .. string.format("%.1f", med))
end

File: 1-iniciante/1040/1040.c
#include <stdio.h>

int main() {
    double n1, n2, n3, n4, n5;
    double med;
    scanf("%lf%lf%lf%lf", &n1, &n2, &n3, &n4);

    med = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / 10;
    printf("Media: %.1lf\n", med);

    if (med >= 7) printf("Aluno aprovado.\n");
    else if (med < 5) printf("Aluno reprovado.\n");
    else if (5 <= med && med < 7) {
        printf("Aluno em exame.\n");

        scanf("%lf", &n5);
        printf("Nota do exame: %.1lf\n", n5);
        
        med = (med + n5) / 2;

        if (med >= 5) printf("Aluno aprovado.\n");
        else if (med < 5) printf("Aluno reprovado.\n");
        printf("Media final: %.1lf\n", med);
    }
    return 0;
}